catchup/service.go

```diff
@@ -868,72 +868,34 @@ func (s *Service) roundIsNotSupported(nextRound basics.Round) bool {
 
 func createPeerSelector(net network.GossipNode, cfg config.Local, pipelineFetch bool) *peerSelector {
 	var peerClasses []peerClass
-	if cfg.EnableCatchupFromArchiveServers {
-		if pipelineFetch {
-			if cfg.NetAddress != "" { // Relay node
-				peerClasses = []peerClass{
-					{initialRank: peerRankInitialFirstPriority, peerClass: network.PeersConnectedOut},
-					{initialRank: peerRankInitialSecondPriority, peerClass: network.PeersPhonebookArchivalNodes},
-					{initialRank: peerRankInitialThirdPriority, peerClass: network.PeersPhonebookArchivers},
-					{initialRank: peerRankInitialFourthPriority, peerClass: network.PeersPhonebookRelays},
-					{initialRank: peerRankInitialFifthPriority, peerClass: network.PeersConnectedIn},
-				}
-			} else {
-				peerClasses = []peerClass{
-					{initialRank: peerRankInitialFirstPriority, peerClass: network.PeersPhonebookArchivalNodes},
-					{initialRank: peerRankInitialSecondPriority, peerClass: network.PeersPhonebookArchivers},
-					{initialRank: peerRankInitialThirdPriority, peerClass: network.PeersConnectedOut},
-					{initialRank: peerRankInitialFourthPriority, peerClass: network.PeersPhonebookRelays},
-				}
+	if pipelineFetch {
+		if cfg.NetAddress != "" && cfg.EnableGossipService { // Relay node
```

Comment:
Since all nodes going to keep 22k blocks back maybe it makes sense to have Out, Archival, Relays, In sequence for all - in a subsequent PR when peer selector will be able to switch between Out/Archival roles faster without exhausting ranks over all Out nodes in hundreds attempts.

Reply:
Agreed, purposesly did not change the ordering as this PR was lasered in on removing the active usage of archivers. Note the addition of the EnableGossipService check here.